crates/miden-objects/src/transaction/script_builder.rs

```diff
@@ -0,0 +1,314 @@
+    /// - The transaction script compilation fails
+    pub fn compile_tx_script(
+        &self,
+        tx_script: &str,
+    ) -> Result<TransactionScript, ScriptBuilderError> {
+        let mut assembler = self.assembler.clone();
+
+        // Add all libraries from the builder to the assembler
+        for lib in &self.libraries {
+            assembler = assembler.with_library(lib.clone()).map_err(|err| {
+                ScriptBuilderError::TransactionScriptLibraryError(alloc::format!(
+                    "Failed to add library: {err}"
+                ))
+            })?;
+        }
+
+        TransactionScript::compile(tx_script, assembler)
+            .map_err(ScriptBuilderError::TransactionScriptError)
+    }
+
+    /// Compiles a note script with the provided program code.
+    ///
+    /// The compiled script will have access to all libraries that have been added to this builder.
+    ///
+    /// # Arguments
+    /// * `program` - The note script source code
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - Any of the libraries cannot be loaded into the assembler
+    /// - The note script compilation fails
+    pub fn compile_note_script(&self, program: &str) -> Result<NoteScript, ScriptBuilderError> {
+        let mut assembler = self.assembler.clone();
+
+        // Add all libraries from the builder to the assembler
+        for lib in &self.libraries {
+            assembler = assembler.with_library(lib.clone()).map_err(|err| {
+                ScriptBuilderError::TransactionScriptLibraryError(alloc::format!(
+                    "Failed to add library: {err}"
+                ))
+            })?;
+        }
+
+        NoteScript::compile(program, assembler).map_err(|err| {
+            ScriptBuilderError::LibraryBuildError(alloc::format!(
+                "Failed to compile note script: {err}"
+            ))
+        })
+    }
+
+    // LIBRARY COMPILATION
+    // --------------------------------------------------------------------------------------------
+
+    /// Compiles the provided library code and library path into a Library.
+    ///
+    /// This method allows you to compile libraries independently and pass them
+    /// to `compile_tx_script()` as needed. This matches the original client API pattern:
+    /// ```ignore
+    /// let library = builder.compile_library(account_code, library_path)?;
+    /// let tx_script = builder.compile_tx_script(script_code, Some(library))?;
+    /// ```
+    ///
+    /// # Arguments
+    /// * `library_code` - The source code of the library
+    /// * `library_path` - The path identifier for the library (e.g., "my_lib::my_module")
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - The library path is invalid
+    /// - The library code cannot be parsed
+    /// - The library cannot be assembled
+    pub fn compile_library(
+        &self,
+        library_code: &str,
+        library_path: &str,
+    ) -> Result<Library, ScriptBuilderError> {
```

Comment:
Similar to the previous two comments, could we use the following signature:
```rust
pub fn compile_library(
    &self,
    library_code: impl AsRef<str>,
    library_path: impl TryInto<LibraryPath>,
) -> Result<Library, ScriptBuilderError>
```
Or does it create some complications?